src/Data/Models/Wallet.cs

```diff
@@ -109,7 +109,7 @@ public class Wallet : Entity
             {
                 var bitcoinExtPubKeys = Keys.Select(x =>
                     {
-                        if (x.InternalWalletId != null)
+                        if (IsHotWallet && x.InternalWalletId != null)
```

Comment:
Maybe here is only the first necessary
```suggestion
                        if (IsHotWallet)
```